src/vs/editor/browser/gpu/raster/raster.ts

```diff
@@ -0,0 +1,59 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import type { MetadataConsts } from '../../../common/encodedTokenAttributes.js';
+
+export interface IGlyphRasterizer {
+	/**
+	 * A unique identifier for the rasterizer.
+	 */
+	id: number;
+
+	/**
+	 * Rasterizes a glyph.
+	 * @param chars The character(s) to rasterize. This can be a single character, a ligature, an
+	 * emoji, etc.
+	 * @param metadata The metadata of the glyph to rasterize. See {@link MetadataConsts} for how
+	 * this works.
+	 * @param colorMap A theme's color map.
+	 */
+	rasterizeGlyph(
+		chars: string,
+		metadata: number,
+		colorMap: string[],
+	): Readonly<IRasterizedGlyph>;
+}
+
+/**
+ * A simple bounding box in a 2D plane.
+ */
+export interface IBoundingBox {
+	/** The left x coordinate (inclusive). */
+	left: number;
+	/** The top y coordinate (inclusive). */
+	top: number;
+	/** The right x coordinate (inclusive). */
+	right: number;
+	/** The bottom y coordinate (inclusive). */
+	bottom: number;
+}
+
+/**
+ * A glyph that has been rasterized to a canvas.
+ */
+export interface IRasterizedGlyph {
+	/**
+	 * The source canvas the glyph was rasterized to.
+	 */
+	source: OffscreenCanvas;
+	/**
+	 * The bounding box of the glyph within {@link source}.
+	 */
+	boundingBox: IBoundingBox;
+	/**
+	 * The offset to the glyph's origin (where it should be drawn to).
+	 */
+	originOffset: { x: number; y: number };
```

Comment:
I see `Point` is in inline edits so it would need to move. But anyway, I think it complicates things using a class here as this will likely need to be serializable eventually and then we would need to handle serialization/deserialization manually, whereas we get it for free currently.